src/invidious/helpers/extractors.cr

```diff
@@ -0,0 +1,562 @@
+      if length_container = item_contents["lengthText"]?
+        length_seconds = decode_length_seconds(length_container["simpleText"].as_s)
+      elsif length_container = item_contents["thumbnailOverlays"]?.try &.as_a.find(&.["thumbnailOverlayTimeStatusRenderer"]?)
+        length_seconds = extract_text(length_container["thumbnailOverlayTimeStatusRenderer"]["text"]).try { |t| decode_length_seconds(t) } || 0
+      else
+        length_seconds = 0
+      end
+
+      live_now = false
+      paid = false
+      premium = false
+
+      premiere_timestamp = item_contents.dig?("upcomingEventData", "startTime").try { |t| Time.unix(t.as_s.to_i64) }
+
+      item_contents["badges"]?.try &.as_a.each do |badge|
+        b = badge["metadataBadgeRenderer"]
+        case b["label"].as_s
+        when "LIVE NOW"
+          live_now = true
+        when "New", "4K", "CC"
+          # TODO
+        when "Premium"
+          # TODO: Potentially available as item_contents["topStandaloneBadge"]["metadataBadgeRenderer"]
+          premium = true
+        else nil # Ignore
+        end
+      end
+
+      SearchVideo.new({
+        title:              title,
+        id:                 video_id,
+        author:             author,
+        ucid:               author_id,
+        published:          published,
+        views:              view_count,
+        description_html:   description_html,
+        length_seconds:     length_seconds,
+        live_now:           live_now,
+        premium:            premium,
+        premiere_timestamp: premiere_timestamp,
+      })
+    end
+  end
+
+  # Parses a InnerTube channelRenderer into a SearchChannel. Returns nil when the given object isn't a channelRenderer
+  #
+  # A channelRenderer renders a channel to click on within the YouTube and Invidious UI. It is **not**
+  # the channel page itself.
+  #
+  # See specs for example.
+  #
+  # `channelRenderer`s can be found almost everywhere on YouTube. In categories, search results, channels, etc.
+  #
+  module ChannelRendererParser
+    def self.process(item : JSON::Any, author_fallback : AuthorFallback)
+      if item_contents = (item["channelRenderer"]? || item["gridChannelRenderer"]?)
+        return self.parse(item_contents, author_fallback)
+      end
+    end
+
+    private def self.parse(item_contents, author_fallback)
+      author = extract_text(item_contents["title"]) || author_fallback.name || ""
+      author_id = item_contents["channelId"]?.try &.as_s || author_fallback.id || ""
+
+      author_thumbnail = HelperExtractors.get_thumbnails(item_contents)
+      # When public subscriber count is disabled, the subscriberCountText isn't sent by InnerTube.
+      # Always simpleText
+      # TODO change default value to nil
+      subscriber_count = item_contents.dig?("subscriberCountText").try &.["simpleText"].try { \
+         |s| short_text_to_number(s.as_s.split(" ")[0]) } || 0
+
+      auto_generated = !item_contents["videoCountText"]? ? true : false
```

Comment:
I'm unsure about what result is expected, but this should be simplified.

Something like [`.nil?`](https://crystal-lang.org/api/1.1.1/Object.html#nil?:Bool-instance-method) is probablly better suited for that purpose (it's a conditional, so it returns a boolean directly, non need for `? :`):

```crystal
# True if key `"videoCountText"` doesn't exist
# False if key exists
item_contents["videoCountText"]?.nil?

# Inverted results
!(item_contents["videoCountText"]?.nil?)
```